src/search-manager/FilterByBlockType.tsx

```diff
@@ -220,6 +224,23 @@ const FilterByBlockType: React.FC<Record<never, never>> = () => {
     setProblemTypesFilter,
   } = useSearchContext();
 
+  useEffect(() => {
+    if (disabled) {
+      // Clear filters when disabled
+      const selectedBlockTypesFilter = blockTypesFilter;
+      const selectedProblemTypesFilter = problemTypesFilter;
+      setBlockTypesFilter([]);
+      setProblemTypesFilter([]);
```

Comment:
```suggestion
      clearFilters();
```

Small nit: Move this below `clearFilters` function and use it.